src/s2geography/geoarrow_test.cc

```diff
@@ -202,3 +208,149 @@ TEST(GeoArrow, GeoArrowReaderReadWKTCollection) {
   EXPECT_EQ(linestring->edge(0).v0, S2LatLng::FromDegrees(1, 0).ToPoint());
   EXPECT_EQ(linestring->edge(0).v1, S2LatLng::FromDegrees(3, 2).ToPoint());
 }
+
+using s2geography::geoarrow::Writer;
+
+TEST(GeoArrow, GeoArrowWriterPoint) {
+  s2geography::WKTReader reader;
+  auto geog1 = reader.read_feature("POINT (0 1)");
+  auto geog2 = reader.read_feature("POINT (2 3)");
+
+  Writer writer;
+  nanoarrow::UniqueSchema schema;
+  nanoarrow::UniqueArray array;
+
+  InitSchemaGeoArrowPoint(schema.get());
+  writer.Init(schema.get());
+
+  writer.WriteGeography(*geog1);
+  writer.WriteGeography(*geog2);
+  writer.Finish(array.get());
+
+  EXPECT_EQ(array->length, 2);
+
+  // EXPECT_THAT(nanoarrow::ViewArrayAs<double_t>(array->children[0]), ElementsAre(DoubleEq(0.0), DoubleEq(2.0)));
+  auto xs = reinterpret_cast<const double*>(array->children[0]->buffers[1]);
+  EXPECT_DOUBLE_EQ(xs[0], 0.0);
+  EXPECT_DOUBLE_EQ(xs[1], 2.0);
+  // EXPECT_THAT(nanoarrow::ViewArrayAs<double_t>(array->children[1]), ElementsAre(DoubleEq(1.0), DoubleEq(3.0)));
+  auto ys = reinterpret_cast<const double*>(array->children[1]->buffers[1]);
+  EXPECT_DOUBLE_EQ(ys[0], 1.0);
+  EXPECT_DOUBLE_EQ(ys[1], 3.0);
+}
+
+TEST(GeoArrow, GeoArrowWriterPointProjected) {
+  s2geography::WKTReader reader;
+  auto geog1 = reader.read_feature("POINT (0 1)");
+  auto geog2 = reader.read_feature("POINT (2 3)");
+
+  Writer writer;
+  nanoarrow::UniqueSchema schema;
+  nanoarrow::UniqueArray array;
+
+  s2geography::geoarrow::ExportOptions options;
+  options.set_projection(s2geography::pseudo_mercator());
+  InitSchemaGeoArrowPoint(schema.get());
+  writer.Init(schema.get(), options);
+
+  writer.WriteGeography(*geog1);
+  writer.WriteGeography(*geog2);
+  writer.Finish(array.get());
+
+  EXPECT_EQ(array->length, 2);
+
+  auto xs = reinterpret_cast<const double*>(array->children[0]->buffers[1]);
+  EXPECT_DOUBLE_EQ(xs[0], 0.0);
+  EXPECT_DOUBLE_EQ(xs[1], 222638.98158654661);
+  auto ys = reinterpret_cast<const double*>(array->children[1]->buffers[1]);
+  EXPECT_DOUBLE_EQ(ys[0], 111325.14286638441);
+  EXPECT_DOUBLE_EQ(ys[1], 334111.17140195851);
+}
+
+TEST(GeoArrow, GeoArrowWriterPolylineTessellated) {
+  s2geography::WKTReader reader;
+  auto geog = reader.read_feature("LINESTRING (-64 45, 0 45)");
+
+  Writer writer;
+  nanoarrow::UniqueSchema schema;
+  GeoArrowSchemaInitExtension(schema.get(), GEOARROW_TYPE_LINESTRING);
```

Comment:
I am essentially using geoarrow-c for this line. For points you wrote the custom `InitSchemaGeoArrowPoint` helper higher up in this file. But if we are going to do this for multiple geoarrow types, might be easier to depend on geoarrow-c? 

(otherwise I can certainly write something similar for Linestring and Polygon ..)

Reply:
> I only tested "point" here because our internal code doesn't special-case any specific geoarrow types (it relies on geoarrow-c's visitor to correctly handle those cases or error).

In practice testing more types did discover a bug on our side (not every type is as picky on correctly calling every visitor method, it seems ;)). Anyway, given we don't yet have that much of tests, the more testing the better for now